src/diffusers/loaders/single_file_model.py

```diff
@@ -430,6 +431,8 @@ def from_single_file(cls, pretrained_model_link_or_path_or_dict: Optional[str] =
                 keep_in_fp32_modules=keep_in_fp32_modules,
                 unexpected_keys=unexpected_keys,
             )
+            empty_device_cache()
+            device_synchronize()
```

Comment:
Maybe having a comment saying why we call before the rest of the code would be helpful.